src/coord/src/coord.rs

```diff
@@ -998,10 +999,47 @@ impl Coordinator {
         }
     }
 
+    fn validate_update_iter(
```

Comment:
Can you add a comment to this method describing the assertions it makes? Pretty much a copy of the commit message.